### ISSUE TITLE
pnpm update pins preview dependencies

### ISSUE THREAD

**Issue by fflaten** [pnpm/pnpm#7002]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.7.0

### Which area(s) of pnpm are affected? (leave empty if unsure)

Dependencies resolver

### Reproduction steps

Create a package.json with the following dependencies:

```
"dependencies": {
    "astro": "^3.0.0-rc.8",
    "satori": "^0.10.0",
  },
```

Run `pnpm up`.

### Describe the Bug

Running `pnpm update` updates dependency specifier in `package.json`, but pins preview-versions.

```
# this is expected
"satori": "^0.10.0" becomes "^0.10.3"
# this is not. why is the caret removed?
"astro": "^3.0.0-rc.8" becomes "3.0.0-rc.11"
```

### Expected Behavior

Ranges are not removed for preview-versions, similar to stable versions. In the sample above `^3.0.0-rc.11` would be expected after the update.

### Which Node.js version are you using?

18.16.0

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [X] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

Ubuntu

**Comment by KSXGitHub** [comment]
I have a feeling that this may be intended behavior.

@zkochan Is this intended?

**Comment by zkochan** [comment]
Using a `^` with prereleased versions is definetely a bad idea. But in this case it should preserve the prefix.

**Comment by KSXGitHub** [comment]
> Using a `^` with prereleased versions is definetely a bad idea. But in this case it should preserve the prefix.

The issue is about the caret being removed. The suffix (I think you mean this instead of "prefix") is already preserved. So I think it's safe to closed this issue as "not planned".

**Comment by zkochan** [comment]
No, I mean the prefix. `^` is the prefix. It should not be removed on `pnpm update`. It should become `^3.0.0-rc.11`, not `3.0.0-rc.11`

**Comment by fflaten** [comment]
> But in this case it should preserve the prefix.

Yes please 👍 

> Using a `^` with prereleased versions is definetely a bad idea. 

As a general rule, yes. In this case the prerelease was stable enough for some non-critical projects, and the caret would allow GA/stable once available. 🙂

**Comment by lkwr** [comment]
Any updates on this?

We often use prereleases for our internal packages, and it's super annoying to have to update these dependencies manually all the time. Previously we used Bun as the package manager and we didn't have this issue.
